core-java-modules/core-java-datetime-string-2/src/test/java/com/baeldung/timestamp/CurrentTimeAsFileNameUnitTest.java

```diff
@@ -0,0 +1,104 @@
+import java.time.temporal.ChronoUnit;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.regex.Pattern;
+
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.junit.Test;
+
+public class CurrentTimeAsFileNameUnitTest {
+    private static final String TIMESTAMP_FORMAT = "yyyyMMddHHmmss";
+    private static final DateTimeFormatter DATETIME_FORMATTER = DateTimeFormatter.ofPattern(TIMESTAMP_FORMAT);
+    private static final SimpleDateFormat SIMPLEDATE_FORMATTER = new SimpleDateFormat(TIMESTAMP_FORMAT);
+
+    private String getFileName(String currentTime) {
+        return MessageFormat.format("{0}.txt", currentTime);
+    }
+
+    private boolean verifyFileName(String fileName) {
+        return Pattern
+          .compile("[0-9]{14}+\\.txt", Pattern.CASE_INSENSITIVE)
+          .matcher(fileName)
+          .matches();
+    }
+
+    @Test
+    public void whenUsingCalender_thenGetCurrentTime() {
+        String currentTime = SIMPLEDATE_FORMATTER.format(Calendar.getInstance().getTime());
+        String fileName = getFileName(currentTime);
+        assertTrue(verifyFileName(fileName));
+    }
+
+    @Test
+    public void whenUsingDate_thenGetCurrentTime() {
+        String currentTime = SIMPLEDATE_FORMATTER.format(new Date());
+        String fileName = getFileName(currentTime);
+        assertTrue(verifyFileName(fileName));
+    }
+    
+    @Test
+    public void whenUsingInstant_thenGetCurrentTime() {
+        String currentTime = Instant
+          .now()
+          .truncatedTo(ChronoUnit.SECONDS)
+          .toString()
+          .replaceAll("[:TZ-]", "");
+        String fileName = getFileName(currentTime);
+        assertTrue(verifyFileName(fileName));
+    }
+
+    @Test
+    public void whenUsingLocalDateTime_thenGetCurrentTime() {
+        String currentTime = LocalDateTime.now().format(DATETIME_FORMATTER);
+        String fileName = getFileName(currentTime);
+        assertTrue(verifyFileName(fileName));
+    }
+
+    @Test
+    public void whenUsingZonedDateTime_thenGetCurrentTime() {
+        String currentTime = ZonedDateTime
+          .now(ZoneId.of("Europe/Paris"))
+          .format(DATETIME_FORMATTER);
+        String fileName = getFileName(currentTime);
+        assertTrue(verifyFileName(fileName));
+    }
+
+    @Test
+    public void whenUsingOffsetDateTime_thenGetCurrentTime() {
+        String currentTime = OffsetDateTime
+          .of(LocalDateTime.now(), ZoneOffset.of("+01:00"))
+          .format(DATETIME_FORMATTER);
+        String fileName = getFileName(currentTime);
+        assertTrue(verifyFileName(fileName));
+    }
+
+    @Test
+    public void whenUsingJodaTime_thenGetCurrentTime() {
+        String currentTime = DateTime.now().toString(TIMESTAMP_FORMAT);
+        String fileName = getFileName(currentTime);
+        assertTrue(verifyFileName(fileName));
+    }
+
+    @Test
+    public void whenUsingJodaTimeInstant_thenGetCurrentTime() {
```

Comment:
Can we rename this to `whenUsingJodaInstant_thenGetCurrentTime`?

Essentially the format would be `whenUsingJoda<CLASS_NAME>`